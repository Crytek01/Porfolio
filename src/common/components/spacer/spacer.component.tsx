import { FunctionComponent } from 'react';

import { Space } from './spacer.styles';

const POSITION_VARIANTS = {
  top: 'margin-top',
  right: 'margin-right',
  bottom: 'margin-bottom',
  left: 'margin-left',
} as const;

interface ISpacer {
  position?: keyof typeof POSITION_VARIANTS;
  size?: number;
  children?: React.ReactNode;
}

export const Spacer: FunctionComponent<ISpacer> = ({
  position = 'bottom',
  size = 4,
  children,
}) => (
  <Space position={POSITION_VARIANTS[position]} size={size}>
    {children}
  </Space>
);
