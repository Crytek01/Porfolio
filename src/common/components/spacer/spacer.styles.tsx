import styled from 'styled-components';

import { spacing } from '../../../infrastructure/theme/spacing';

interface ISpace {
  position: string;
  size: number;
}

const getStyle = (position: string, size: number): string =>
  `${position}: ${spacing[size]}`;

export const Space = styled.div<ISpace>`
  ${({ position, size }) => getStyle(position, size)}
`;
