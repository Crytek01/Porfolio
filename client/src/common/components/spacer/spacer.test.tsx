import { shallowRender } from '@common/utils/test/shallowRender';
import { describe, expect, it } from 'vitest';

import { Spacer } from './spacer.component';

describe('Spacer component Test', () => {
  it('renders correctly', () => {
    const renderer = shallowRender(<Spacer position="bottom" size={10} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
