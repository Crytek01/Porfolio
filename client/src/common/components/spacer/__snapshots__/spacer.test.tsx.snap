// Vitest Snapshot v1

exports[`Spacer component Test > renders correctly 1`] = `
<styled.div
  position="margin-bottom"
  size={10}
/>
`;
